components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/SAMLSSOConfigServiceImpl.java

```diff
@@ -96,6 +97,23 @@ public SAMLSSOServiceProviderDTO createServiceProvider(SAMLSSOServiceProviderDTO
 
         try {
             SAMLSSOConfigAdmin configAdmin = new SAMLSSOConfigAdmin(getConfigSystemRegistry());
+            if (!StringUtils.isNotBlank(spDto.getSigningAlgorithmURI())
+                    || !Arrays.asList(getSigningAlgorithmUris()).contains(spDto.getSigningAlgorithmURI())) {
+                throw buildClientException(INVALID_REQUEST, "Invalid Response Signing Algorithm.");
```

Comment:
shall we log the signing algorithm in the error message? e.g "Invalid Response Signing Algorithm :" + spDto.getSigningAlgorithmURI().
we can follow this for all the below scenarios.

Reply:
addressed in 4218e66f5163332a0be366598fd467c19187e8a5